code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -718,7 +718,7 @@
 	M.adjustToxLoss(-0.5, 0)
 	M.adjustOxyLoss(-0.5, 0)
 	if(M.nutrition && (M.nutrition - 2 > 0))
-		if(M.mind?.assigned_role != JOB_NAME_MEDICALDOCTOR) //Drains the nutrition of the holder. Not medical doctors though, since it's the Doctor's Delight!
+		if(M.mind && !HAS_TRAIT(M.mind, TRAIT_MEDICAL_METABOLISM)) //Drains the nutrition of the holder. Not medical staff though, since it's the Doctor's Delight!
```

Comment:
Just check if its in GLOB.medical_positions, you really don't need a mind trait for this

Reply:
GLOB.medical_positions doesn't include the Psychiatrist sadly. Also I want it to function like security having LAW_ENFORCEMENT_METABOLISM and all the foods having special effects for sec. Plus, this way you get to keep the benefits even if you change your job.

Reply:
Fair enough then aye, feel free to resolve this